File: gridgraph/tools/convert.py
import struct
with open('/home/junfeng/cit-Patents/cit-Patents.v', 'r') as f:
  vs = {}
  id = 1
  for line in f:
    vs[int(line.split()[0])] = id
    id += 1
  print(f'max id: {id}')
  with open('/home/junfeng/cit-Patents/cit-Patents.e', 'r') as f:
    with open('../data/citpatents', 'wb+') as f1:
      for line in f:
        u, v = map(int, line.split())
        uid = vs[u]
        vid = vs[v]
        f1.write(struct.pack('<l', int(uid)))
        f1.write(struct.pack('<l', int(vid)))

File: gridgraph/tools/convert.cc
#include <fstream>
#include <iostream>
#include <fcntl.h>
#include <unistd.h>

int main(int argc, char** argv) {
  std::fstream file("/home/junfeng/cit-Patents/cit-Patents.e");
  int fd = open("../data/citpatents", O_WRONLY | O_CREAT);
  int edge_count = 0;
  int c = 1;
  int64_t write_bytes = 0;
  while (!file.eof()) {
    int32_t src, dst;
    file >> src >> dst;
    edge_count++;
    write_bytes += write(fd, &src, sizeof(src));
    write_bytes += write(fd, &dst, sizeof(dst));
  }
  std::cout << "write bytes: " << write_bytes << std::endl;
  std::cout << "edge count: " << edge_count << std::endl;
  close(fd);
  return 0;
}